### PR TITLE
Milestone: OOP school library: set up associations

### PR DESCRIPTION
**Pull request by Lancelot-SO**
### The following updates were made: 
- Created a class Classroom with the following:

- [ ] @label instance variable, should be initialized in the constructor.
- [ ] Setter and getter for @label (remember about attr_accessor).

- [ ] Create the has-many/belongs-to relationship between Classroom and Student. The following should be implemented:
- [ ] Create the has-many side (a classroom has many students).
- [ ] Create the belongs-to side (a student belongs to a classroom).
- [ ] Make sure that when adding a student to a classroom it also sets the classroom for the student.
- [ ] Make sure that when setting the classroom for a student it also adds it to the classrooms' students.

- Created a class Book with the following:

- [ ] @title and @author instance variables, should be initialized in the constructor.
- [ ] Setters and getters for instance variables (remember about attr_accessor).

- Created a class Rental with the following:

- [ ] @date instance variable, should be initialized in the constructor.
- [ ] Setter and getter for @date (remember about attr_accessor).
- [ ] Create the many-to-many (also has-many-through) relationship between Person and Book using the intermediate class Rental. The following should be implemented:
- [ ] Create the has-many side of Book and Rental (a book has many rentals).
- [ ] Create the belongs-to side of Rental and Book (a rental belongs to a book).
- [ ] Create the has-many side of Person and Rental (a person has many rentals).
- [ ] Create the belongs-to side of Rental and Person (a rental belongs to a person).
- [ ] Modify the constructor of Rental so Book and Person are set in it.
- Modified readme file.

### CODE REVIEW

File: book.rb [hunk 1]
```diff
@@ -0,0 +1,9 @@
+class Book
+  attr_accessor :title, :author, :rentals
+
+  def initialize(title, author)
+    @title = title
+    @author = author
+    @rentals = []
+  end
+end
```
Comment (Munsa1) [inline book.rb]: - [ ] Kindly add a method for adding rentals to the book class, the book itself should be sent as a parameter to create the new rental achieving the 'has-many' association. e.g.
```
ruby
def add_rental(person, date)
    Rental.new(date, self, person)
  end
```

File: person.rb [hunk 2]
```diff
@@ -5,14 +5,15 @@ def initialize(age, name = 'Unknown', parent_permission: true)
...
 
   def correct_name
     name
   end
 
-  attr_accessor :name, :age
+  attr_accessor :name, :age, :rentals
   attr_reader :id
 
   private
```
Comment (Munsa1) [inline person.rb]: - [ ] A method for adding rentals should be present. The person itself should be sent as a parameter to create the new rental achieving the 'has-many' association.

**Review by Munsa1 — CHANGES_REQUESTED** [review]
Hi @Lancelot-SO, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

## Highlights 
Descriptive ReadMe File :heavy_check_mark: 

## Required Changes ♻️
_Check the comments under the review._
## Due to insufficient files changed I will leave some comments here

- [ ] Kindly implement the he 'belongs-to' side (a student belongs to a classroom). Also add `attr_reader :classroom` to the student class 
image
code snippet extample
```
def classroom=(classroom)
      @classroom = classroom
      classroom.students.push(self) unless classroom.students.include?(self)
  end
end
```



## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- x
- y
- z

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by tresorsawasawa — COMMENTED** [review]
# STATUS: APPROVE :heavy_check_mark: :heavy_check_mark:  :champagne: 

Hi @Lancelot-SO 
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 :clap: :clap: :100: 

nodding-yes

## To highlight :champagne: 
- All linters are passing :heavy_check_mark: 
- Good commit history :heavy_check_mark: 
- Good PR description :heavy_check_mark: 
- Your README file is professional :heavy_check_mark: 

## Optional suggestions
_Every comment with the **[OPTIONAL]** prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

_Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please ping me [@tresorsawasawa](https://github.com/tresorsawasawa) when you comment so I can receive the notification or use slack with the same name._

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**